### ISSUE TITLE
Include specialised versions of Poker.Range in poker-base

### ISSUE THREAD

**Issue by santiweight** [tonyday567/poker-fold#2]
`ShapedHandRange`. We can definitely accommodate this in `poker-base`. This might also be a good opportunity to use `newtype ShapedHand = ShapedHand Word8` for `ShapedHand` since I think with pattern synonyms we can hide this implementation detail...

```
newtype ShapedHandS = ShapedHandS {unShapedHandS :: Word8}

newtype RangedHand a = RangedHand
  { array :: Array '[169] a
  }
```

The rest of the functions in `ShapedHandRange` seem very reasonable and easy to add to `poker-base`.

Design question - should we use data/type families for this? I think we should just leave it for now and include a super-pretty API if we really need to... (We have agreed this in private also I believe)

I couldn't find any other specialised Ranges yet - let me know if I missed one!

**Comment by tonyday567** [comment]
Yes, RangedHand is the only specialisation there. An issue is that Array picks up a dependency on numhask-array which comes with some complexity. I just love using the representable methods - tabulate and index makes so much sense once you get used to them and create compact clear code, but it isn't standard by any stretch, so it might cause confusion for others.
